src/main/java/htsjdk/samtools/cram/io/CramInt.java

```diff
@@ -40,7 +40,10 @@ public static int int32(final byte[] data) {
      * @return an integer value read from the buffer
      */
     public static int int32(final ByteBuffer buffer) {
-        return buffer.get() | buffer.get() << 8 | buffer.get() << 16 | buffer.get() << 24;
+        return (0xFF & (int) buffer.get()) |
+                (0xFF & (int) buffer.get()) << 8 |
+                (0xFF & (int) buffer.get()) << 16 |
+                (0xFF & (int) buffer.get()) << 24;
```

Comment:
While we're in this class I'd be in favor of renaming the `int32` overloads in  to `readInt32` so they're symmetric with the `writeInt32` ones. WDYT ?

Reply:
I like that